### PR TITLE
feat: Model of the project, Lateral line and Derivation line

### PR DESCRIPTION
**Pull request by evandrosystems**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -21,17 +22,17 @@ pluggy==1.2.0
...
+SQLAlchemy==2.0.18
+SQLAlchemy-Utils==0.41.1
 starlette==0.27.0
 toml==0.10.2
 tomli==2.0.1
-typing_extensions==4.7.1
+typing-extensions==4.7.1
 urllib3==2.0.3
 uvicorn==0.22.0
-sqlalchemy==2.0.18
```
Comment (ravellys) [inline requirements.txt:35]: Why did you remove these dependencies?

File: irrigation_design_system_backend/infrastructure/persistence/models/projects.py [hunk 2]
```diff
@@ -1,9 +1,51 @@
...
-    user_id = Column(Integer, nullable=False)
+    __tablename__ = "projects"
+
+    id = Column(BigInteger, primary_key=True, index=True, autoincrement=True)
+    user_id = Column(BigInteger, index=True)
+    group_id = Column(String(100))
+    status = Column(String(100))
+    crop = Column(String(100))
+    irn_max = Column(DECIMAL(10, 2), default=0.00)
+    etc = Column(DECIMAL(10, 2), default=0.00)
```
Comment (ravellys) [inline irrigation_design_system_backend/infrastructure/persistence/models/projects.py:16]: put the full name of the columns: irn, etc, itn, ... please

File: irrigation_design_system_backend/infrastructure/persistence/models/projects.py [hunk 3]
```diff
@@ -1,9 +1,51 @@
...
+    itn = Column(DECIMAL(10, 2), default=0.00)
+
+    # Foreign Keys
+    lateral = relationship('LateralLine', back_populates='project', uselist=True)
+
```
Comment (ravellys) [inline irrigation_design_system_backend/infrastructure/persistence/models/projects.py:21]: you forgout the derivationLine FK

File: irrigation_design_system_backend/infrastructure/persistence/models/projects.py [hunk 4]
```diff
@@ -1,9 +1,51 @@
...
 
 
 class Project(Base):
-    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
-    user_id = Column(Integer, nullable=False)
+    __tablename__ = "projects"
+
+    id = Column(BigInteger, primary_key=True, index=True, autoincrement=True)
+    user_id = Column(BigInteger, index=True)
+    group_id = Column(String(100))
```
Comment (ravellys) [inline irrigation_design_system_backend/infrastructure/persistence/models/projects.py:12]: we need a column with the project description too

**Review by ravellys — CHANGES_REQUESTED** [review]
Some requests before the approve

File: irrigation_design_system_backend/infrastructure/persistence/models/projects.py (hunk 4 above)
Comment (evandrosystems) [reply irrigation_design_system_backend/infrastructure/persistence/models/projects.py:12]: Right

File: requirements.txt (hunk 1 above)
Comment (evandrosystems) [reply requirements.txt:35]: I did a pull in main, and then came changes in requirements.txt, but here the dependencies are installed.

File: irrigation_design_system_backend/infrastructure/persistence/models/projects.py (hunk 2 above)
Comment (evandrosystems) [reply irrigation_design_system_backend/infrastructure/persistence/models/projects.py:16]: Right!

File: irrigation_design_system_backend/infrastructure/persistence/models/projects.py (hunk 3 above)
Comment (evandrosystems) [reply irrigation_design_system_backend/infrastructure/persistence/models/projects.py:21]: Perfect! I will make the relationship with the design table

File: irrigation_design_system_backend/infrastructure/persistence/models/projects.py [hunk 5]
```diff
@@ -1,9 +1,57 @@
-from sqlalchemy import Column, Integer
+from sqlalchemy import Column, String, DECIMAL, ForeignKey, BigInteger, Enum, JSON
+from sqlalchemy.orm import relationship
 
 from infrastructure.persistence.models.base import Base
+from core.domain.enum.line_types import LineTypes
 
 
 class Project(Base):
-    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
```
Comment (ravellys) [inline irrigation_design_system_backend/infrastructure/persistence/models/projects.py:7]: here, add the ENum status. 

thes status are:
DRAFT, ACTIVE, INACTIVE

**Review by ravellys — CHANGES_REQUESTED** [review]

File: Makefile
Comment (ravellys) [inline Makefile:1]: Awesome,

Just put a comment explaining the command and add this on the Readme.md

Comment (ravellys) [reply Makefile:1]: You can create the make requirements too

**Review by ravellys — CHANGES_REQUESTED** [review]

File: Makefile
Comment (evandrosystems) [reply Makefile:1]: Right!

**Review by ravellys — APPROVED** [review]
Awesome

**Pull request merged**